churaverse-plugins-client/src/churarenPlugin/churarenBossPlugin/controller/socketController.ts

```diff
@@ -0,0 +1,95 @@
+  Position,
+  EntitySpawnEvent,
+  EntityDespawnEvent,
+  LivingDamageEvent,
+} from 'churaverse-engine-client'
+import { RegisterMessageEvent } from '@churaverse/network-plugin-client/event/registerMessageEvent'
+import { RegisterMessageListenerEvent } from '@churaverse/network-plugin-client/event/registerMessageListenerEvent'
+import { BaseSocketController } from '@churaverse/network-plugin-client/interface/baseSocketController'
+import { BossPluginStore } from '../store/defBossPluginStore'
+import { BossSpawnMessage } from '../message/bossSpawnMessage'
+import { BossDespawnMessage } from '../message/bossDespawnMessage'
+import { BossWalkMessage } from '../message/bossWalkMessage'
+import { Boss } from '../domain/boss'
+import { WeaponDamageMessage } from '@churaverse/player-plugin-client/message/weaponDamageMessage'
+import { CollisionBossDamageCause } from '../domain/collisionBossDamageCause'
+import { BossWalkEvent } from '../event/bossWalkEvent'
+import { IMessageListenerRegister } from '@churaverse/network-plugin-client/interface/IMessageListenerRegister'
+
+export class SocketController extends BaseSocketController<IMainScene> {
+  public bossPluginStore!: BossPluginStore
+  public messageListenerRegister!: IMessageListenerRegister<IMainScene>
+
+  public constructor(eventBus: IEventBus<IMainScene>, store: Store<IMainScene>) {
+    super(eventBus, store)
+  }
+
+  public registerMessage(ev: RegisterMessageEvent<IMainScene>): void {
+    ev.messageRegister.registerMessage('bossSpawn', BossSpawnMessage, 'queue')
+    ev.messageRegister.registerMessage('bossDespawn', BossDespawnMessage, 'queue')
+    ev.messageRegister.registerMessage('bossWalk', BossWalkMessage, 'queue')
+  }
+
+  public setupMessageListenerRegister(ev: RegisterMessageListenerEvent<IMainScene>): void {
+    this.messageListenerRegister = ev.messageListenerRegister
+  }
+
+  public registerMessageListener(): void {
+    this.messageListenerRegister.on('bossSpawn', this.bossSpawn)
+    this.messageListenerRegister.on('bossDespawn', this.bossDespawn)
+    this.messageListenerRegister.on('weaponDamage', this.collisionBossDamage)
+    this.messageListenerRegister.on('bossWalk', this.bossWalk)
+  }
+
+  public unregisterMessageListener(): void {
+    this.messageListenerRegister.off('bossSpawn', this.bossSpawn)
+    this.messageListenerRegister.off('bossDespawn', this.bossDespawn)
+    this.messageListenerRegister.off('weaponDamage', this.collisionBossDamage)
+    this.messageListenerRegister.off('bossWalk', this.bossWalk)
+  }
+
+  public getStore(): void {
+    this.bossPluginStore = this.store.of('bossPlugin')
+  }
+
+  private readonly bossSpawn = (msg: BossSpawnMessage): void => {
+    const data = msg.data
+    const pos = new Position(data.startPos.x, data.startPos.y)
+    const boss = new Boss(data.bossId, pos, data.spawnTime, data.bossHp)
+    if (this.bossPluginStore.bosses.get(data.bossId) !== undefined) return
+    const bossSpawnEvent = new EntitySpawnEvent(boss)
```

Comment:
ブロックだけでなく、行の粒度でもスコープがなるべく小さくなるような順番を意識した方が良いですね
```suggestion
    const data = msg.data
    if (this.bossPluginStore.bosses.get(data.bossId) !== undefined) return
    const pos = new Position(data.startPos.x, data.startPos.y)
    const boss = new Boss(data.bossId, pos, data.spawnTime, data.bossHp)
    const bossSpawnEvent = new EntitySpawnEvent(boss)
```